topdown/object_test.go

```diff
@@ -131,6 +133,24 @@ func TestBuiltinObjectUnion(t *testing.T) {
 			objectB:  `{"a": {"b": "foo", "b1": "bar"}, "d": 7, "e": 17}`,
 			expected: `{"a": {"b": "foo", "b1": "bar"}, "d": 7, "e": 17}`,
 		},
+		{
+			note:     "error wrong lhs type",
+			objectA:  `[1, 2, 3]`,
+			objectB:  `{"b": 2}`,
+			expected: ast.NewError(ast.TypeErr, nil, "object.union: invalid argument(s)"),
+		},
+		{
+			note:     "error wrong rhs type",
+			objectA:  `{"a": 1}`,
+			objectB:  `[1, 2, 3]`,
+			expected: ast.NewError(ast.TypeErr, nil, "object.union: invalid argument(s)"),
+		},
+		{
+			note:     "error wrong both params",
+			objectA:  `"foo"`,
+			objectB:  `[1, 2, 3]`,
+			expected: ast.NewError(ast.TypeErr, nil, "object.union: invalid argument(s)"),
+		},
```

Comment:
We should test the topdown built-in error handling as opposed to the compiler's type checking here. The way to do that is to execute the test with some input data for which the type is not known.

Reply:
Done